crates/egui_demo_lib/src/demo/pan_zoom.rs

```diff
@@ -0,0 +1,106 @@
+#[derive(Clone, Default, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct PanZoom {
+    pan: egui::Vec2,
+    zoom: f32,
+}
+
+impl Eq for PanZoom {}
+
+impl super::Demo for PanZoom {
+    fn name(&self) -> &'static str {
+        "🗖 Pan Zoom"
+    }
+
+    fn show(&mut self, ctx: &egui::Context, open: &mut bool) {
+        use super::View as _;
+        let window = egui::Window::new("Pan Zoom")
+            .default_width(200.0)
+            .default_height(200.0)
+            .vscroll(false)
+            .open(open);
+        window.show(ctx, |ui| self.ui(ui));
+    }
+}
+
+impl super::View for PanZoom {
+    fn ui(&mut self, ui: &mut egui::Ui) {
+        // On initialization, zoom is 0
+        if self.zoom == 0.0 {
+            self.zoom = 1.0;
+        }
+
+        let (id, rect) = ui.allocate_space(ui.available_size());
+        let response = ui.interact(rect, id, egui::Sense::click_and_drag());
+        // Uncomment to allow dragging the background as well.
+        // self.pan += response.drag_delta() / self.zoom;
+
+        // Plot-like reset
+        if response.double_clicked() {
+            self.zoom = 1.0;
+            self.pan = egui::Vec2::ZERO;
+        }
+
+        if let Some(pointer) = ui.ctx().input(|i| i.pointer.hover_pos()) {
+            // Ignore if some other widget is covering this container.
+            if response.rect.contains(pointer) {
+                let original_zoom = self.zoom;
+                self.zoom *= ui.ctx().input(|i| i.zoom_delta());
+                let delta = pointer / self.zoom - pointer / original_zoom;
+                self.pan += delta;
+
+                // Keep mouse centered.
+                self.pan += ui.ctx().input(|i| i.raw_scroll_delta) / self.zoom;
```

Comment:
why not `smooth_scroll_delta`?

Reply:
Ah, I was trying to match the feel of blender's zooming in and out. I guess `smooth_scroll_delta` is better.